prql-compiler/src/sql/gen_expr.rs

```diff
@@ -184,6 +183,60 @@ pub(super) fn translate_literal(l: Literal, ctx: &Context) -> Result<sql_ast::Ex
     })
 }
 
+fn translate_datetime_literal(
+    data_type: sql_ast::DataType,
+    value: String,
+    ctx: &Context,
+) -> sql_ast::Expr {
+    if ctx.dialect.is::<crate::sql::dialect::SQLiteDialect>() {
+        translate_datetime_literal_with_sqlite_function(data_type, value)
+    } else {
+        translate_dateatime_literal_with_typed_string(data_type, value)
+    }
+}
+
+fn translate_dateatime_literal_with_typed_string(
```

Comment:
```suggestion
fn translate_datetime_literal_with_typed_string(
```

(unless this was intentional)